Sources/MongoSwift/Operations/RunCommandOperation.swift

```diff
@@ -35,15 +40,53 @@ internal struct RunCommandOperation: Operation {
     private let database: MongoDatabase
     private let command: BSONDocument
     private let options: RunCommandOptions?
+    private let serverAddress: ServerAddress?
 
-    internal init(database: MongoDatabase, command: BSONDocument, options: RunCommandOptions?) {
+    internal init(
+        database: MongoDatabase,
+        command: BSONDocument,
+        options: RunCommandOptions?,
+        serverAddress: ServerAddress? = nil
+    ) {
         self.database = database
         self.command = command
         self.options = options
+        self.serverAddress = serverAddress
     }
 
     internal func execute(using connection: Connection, session: ClientSession?) throws -> BSONDocument {
```

Comment:
I added a `runCommand` overload that allows commands be sent to specific servers based on address to be used from the tests. This allows us to reuse existing clients that are connected to muliple mongoses to route failpoints to individual servers, instead of creating clients that are only connected to individual servers themselves.

Reply:
That seems like something you'd want to decide in a different piece of work, it's not really "upgrading libmongoc" right?

Reply:
>That seems like something you'd want to decide in a different piece of work, it's not really "upgrading libmongoc" right? I'm not sure you would come up with this solution in isolation, maybe you would want to use a custom server selector for instance.

This overload is testing-only and not exposed to the public API. I agree that a comprehensive solution is preferable and that what is currently implemented would not be it, but this is just a temporary measure to make these tests to run faster in the meantime, since we couldn't merge the libmongoc upgrade otherwise.

>I'm also not sure how to comment on your other comment above (re: closing sync clients), but it's likely that the massive gains you saw in your testing could also relate to reducing the hardcoded minHeartbeatFrequencyMS to 50ms. 

Lowering to 50ms certainly did help, but the tests were still timing out. Reducing client re-usage got them to run without timing out.

>How were you measuring the deinit of the client? 0.5s seems like a very long time, and probably indicative of a bug, rather than necessitating sharing clients across tests.

I just created `Date`s before and after running the `deinit`. From looking further at individual statements in the deinit, it seems the culprit is `mongoc_client_pool_destroy`, and it taking that long could certainly be a bug. I need to look into it some more.

Also note, we're not sharing clients across tests, we're now just reusing clients within individual tests.